### PR TITLE
Tioga host configs based on common install

### LINKED ISSUE #352: Add Tioga host configs

### PR DESCRIPTION
**Pull request by tpatki**
# Description

Fixes #352 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature/architecture support (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Build/CI update

# How Has This Been Tested?

Tested on Tioga. 

# Checklist:

- [ ] I have run `./scripts/check-code-format.sh` and confirm my C/C++ code follows the style guidelines of variorum
- [ ] I have run `flake8` and `black --check --diff` and confirm my python code follows the style guidelines of variorum
- [ ] I have run `./scripts/check-rst-format.sh` and confirm my documentation follows the style guidelines of variorum
- [ ] I have added comments in my code
- [ ] My changes generate no new warnings (build with `-DENABLE_WARNINGS=ON`)
- [ ] New and existing unit tests pass with my changes

Thank you for taking the time to contribute to Variorum!

### CODE REVIEW

**Comment by slabasan** [comment]
@tpatki I get these warnings with CPU-only build and CPU+GPU build:
```
/opt/cray/pe/cce/14.0.4/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: warning: libfi.so.1, needed by /usr/workspace/msr/tioga_common_setup/esmi_install/e_smi/lib/libe_smi64.so, not found (try using -rpath or -rpath-link)
```

**Comment by slabasan** [comment]
@tpatki I'm not sure how we should handle this external library, seems like we want to call a `find_library` in variorum to resolve this. Any thoughts on how to add this to the host config instead of needing to set `LD_LIBRARY_PATH`?

**Comment by tpatki** [comment]
Couldn't really think of a good way @slabasan other than the `find_library` way. 
We can do what we did for other libraries such as `JANSSON_DIR` or `ESMI_DIR`, where we add a SetupCrayLibraries.cmake and then create a `CRAY_LIB_DIR` that we can set in the host config files for the Tioga build. Thoughts?

File: host-configs/tioga-4.18.0toss.t4.x86_64-gcc@10.3.1-rocm@5.2-both.cmake [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Copyright 2019-2022 Lawrence Livermore National Security, LLC and other
```
Comment (slabasan) [inline host-configs/tioga-4.18.0toss.t4.x86_64-gcc@10.3.1-rocm@5.2-both.cmake:1]: Update copyright year to 2023

**Pull request merged**